stories/__tests__/SQFormCheckboxGroup.stories.test.js

```diff
@@ -0,0 +1,191 @@
+import React from 'react';
+import {render, screen, waitFor} from '@testing-library/react';
+import {composeStories} from '@storybook/testing-react';
+import userEvent from '@testing-library/user-event';
+import * as stories from '../SQFormCheckboxGroup.stories';
+
+const {
+  Default: SQFormCheckboxGroup,
+  WithValidation: SQFormCheckboxGroupWithValidation
+} = composeStories(stories);
+
+const SHOPPING_LIST_OPTIONS = [
+  {label: 'Voltfruit', value: 'voltfruit'},
+  {label: 'Sunshroom', value: 'sunshroom'},
+  {label: 'Bokoblin Guts', value: 'bokoblin guts'},
+  {label: 'Lynel Hoof', value: 'lynel hoof'},
+  {label: 'Stealthfin Trout', value: 'stealthfin trout'}
+];
+
+describe('SQFormCheckboxGroup Tests', () => {
+  describe('Checkboxes Only', () => {
+    it('should render a label and a list of checkboxes', () => {
+      render(<SQFormCheckboxGroup size="auto" />);
+
+      const groupLabel = screen.getByText('Shopping List');
+      expect(groupLabel).toBeInTheDocument();
+
+      const checkboxes = screen.getAllByRole('checkbox');
+      expect(checkboxes.length).toBe(SHOPPING_LIST_OPTIONS.length);
+    });
+
+    it('should display as checked when initial values are set', () => {
+      const initialValues = {
+        shoppingList: [SHOPPING_LIST_OPTIONS[0].value]
+      };
+
+      render(<SQFormCheckboxGroup SQFormProps={{initialValues}} size="auto" />);
+
+      const checkboxLabel = screen.getByLabelText(
+        SHOPPING_LIST_OPTIONS[0].label
+      );
+
+      expect(checkboxLabel).toBeChecked();
+
+      const nextCheckboxLabel = screen.getByLabelText(
+        SHOPPING_LIST_OPTIONS[1].label
+      );
+      expect(nextCheckboxLabel).not.toBeChecked();
```

Comment:
I think this one could be a bit more explicit in checking which checkboxes are or are not checked. The first and second checkboxes are being tested, but we don't know for sure that the other checkboxes in the group are behaving as expected. 

So, one option would be to get all of the checkboxes, filter it down to only the checkboxes that are 'checked', then make an assertion on what should be left in that array.

```js
const initialValues = {
  shoppingList: [SHOPPING_LIST_OPTIONS[0].value]
};

render(<SQFormCheckboxGroup SQFormProps={{initialValues}} size="auto" />);

const checkboxes = screen.getAllByRole('checkbox');

const targetCheckbox = screen.getByLabelText(
  SHOPPING_LIST_OPTIONS[0].label
);

expect(checkboxes.filter(checkbox => checkbox)).toEqual(
  expect.arrayContaining([targetCheckbox])
);
```